src/component/application/ApplicationUpdate.tsx

```diff
@@ -1,29 +1,48 @@
-import React, { useState } from 'react';
-import PropTypes from 'prop-types';
+import { useState } from 'react';
 import { TextField, Grid } from '@material-ui/core';
 import { useCommonStyles } from '../../common.styles';
 import icons from './icon-names';
 import GeneralSelect from '../common/GeneralSelect/GeneralSelect';
+import useApplicationsApi from '../../hooks/api/actions/useApplicationsApi/useApplicationsApi';
 
-function ApplicationUpdate({ application, storeApplicationMetaData }) {
+interface IApplication {
+    appName: string;
+    color: string;
+    createdAt: string;
+    description: string;
+    icon: string;
+    instances: [];
+    links: object;
+    seenToggles: [];
+    strategies: [];
+    url: string;
+}
+interface IApplicationUpdateProps {
+    application: IApplication;
+}
+
+const ApplicationUpdate = ({ application }: IApplicationUpdateProps) => {
+    const { storeApplicationMetaData } = useApplicationsApi();
     const { appName, icon, url, description } = application;
-    const [localUrl, setLocalUrl] = useState(url);
-    const [localDescription, setLocalDescription] = useState(description);
+    const [localUrl, setLocalUrl] = useState(url || '');
```

Comment:
Won't this only work if application is always present? Meaning you get to this page via the application list.